packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2754,6 +2758,30 @@ class InputDecoration {
   ///  * [suffixIcon], which is the same but on the trailing edge.
   final Widget prefixIcon;
 
+  /// The constraints for the prefix icon.
+  ///
+  /// This can be used to modify the [BoxConstraints] surrounding [prefixIcon].
+  ///
+  /// This property is particularly useful for getting the decoration's height
+  /// less than 48px. This can be achieved by setting [isDense] to true and
+  /// setting the constraints' minimum height and width to a value lower than
+  /// 48px.
+  ///
+  /// ```dart
```

Comment:
I decided to give it a whirl since I've never tried it :)

Reply:
Although I noticed that testing the new feature in a new DartPad instance didn't work because the parameters did not exist on whatever version of Flutter that DartPad, so we'll have no guarantee that it'll work via the instructions in `snippets/README.md`

Reply:
It will work once DartPad picks up the changes, which should happen in a couple of weeks or so.